### PR TITLE
Persist can return what it has and append a new block.

### PR DESCRIPTION
**Pull request by jacekbilski**

### CODE REVIEW

File: SharpestChain/SharpestChain/data/Block.cs [hunk 1]
```diff
@@ -32,5 +32,47 @@ public static Block fromJson(string json)
         {
             return JsonConvert.DeserializeObject<Block>(json);
         }
+
+        public bool Equals(Block that)
```
Comment (programming-wolf) [inline SharpestChain/SharpestChain/data/Block.cs]: You could add the following:

```
public override bool Equals (object block) 
{
    return base.Equals(obj) || (block instanceof Block ? this.Equals((Block) block) : false);
}
```
In addition, you can then remove all the ReferenceEquals stuff from the second method.

**Comment by jacekbilski** [comment]
Fixed as proposed.

File: SharpestChain/SharpestChain/IO/Persistence.cs [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
     using data;
 
     public class Persistence
     {
-        private static Block[] blocks;
+        public static readonly Block GENESIS_BLOCK = 
+                new Block(1, 0, 1917336, new []{new Transaction(new Guid("b3c973e2-db05-4eb5-9668-3e81c7389a6d"), 0, "I am Heribert Innoq")}, "0");
         
-        public static Block[] Load()
+        private Block[] _blocks = {GENESIS_BLOCK};
```
Comment (programming-wolf) [inline SharpestChain/SharpestChain/IO/Persistence.cs]: `private List<Block> _blocks = new List<Block>();`

File: SharpestChain/SharpestChain/data/Block.cs [hunk 3]
```diff
@@ -32,5 +32,32 @@ public static Block fromJson(string json)
         {
             return JsonConvert.DeserializeObject<Block>(json);
         }
+        
+        public override bool Equals (object block) 
+        {
+            return base.Equals(block) || block is Block && Equals((Block) block);
```
Comment (programming-wolf) [inline SharpestChain/SharpestChain/data/Block.cs]: I think we should invert the check, or even improve it altogether:

```
var other = block as Block;

if (other == null) {
    // it is not a Block, so definitely not equal!
    return false;
}
else
{
 return this == other || Equals(other);
}
```

**Pull request merged**